sdk/storage/Azure.Storage.Blobs/src/PartitionedDownloader.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+
+namespace Azure.Storage.Blobs
+{
+    internal class PartitionedDownloader
+    {
+        private readonly BlobBaseClient _client;
+
+        private readonly long _singleBlockThreshold;
+
+        private readonly int _blockSize;
+        private readonly int _workerCount;
+
+        public PartitionedDownloader(BlobBaseClient client, StorageTransferOptions transferOptions = default, long? singleBlockThreshold = null)
+        {
+            _client = client;
+            _singleBlockThreshold = singleBlockThreshold ?? Constants.Blob.Block.MaxUploadBytes;
+            _workerCount =
+                transferOptions.MaximumConcurrency ?? Constants.Blob.Block.DefaultConcurrentTransfersCount;
+            _blockSize =
+                Math.Min(
+                    Constants.Blob.Block.MaxStageBytes,
+                    transferOptions.MaximumTransferLength ?? Constants.DefaultBufferSize
+                );
+        }
+
+        private static long ParseLength(string s)
+        {
+            int lengthSeparator = s.IndexOf("/", StringComparison.InvariantCultureIgnoreCase);
+
+            if (lengthSeparator == -1)
+            {
+                Errors.ParsingHttpRangeFailed();
+            }
+
+            return long.Parse(s.Substring(lengthSeparator + 1), CultureInfo.InvariantCulture);
+        }
+
+        public async Task<Response<BlobProperties>> DownloadToAsync(Stream stream, BlobRequestConditions conditions, CancellationToken cancellationToken)
```

Comment:
Let's just return `Response` as discussed instead of trying to jam a download into properties.